message_test.go

```diff
@@ -53,3 +58,388 @@ func TestMessageThreadID(t *testing.T) {
+		t.Errorf("msg.Filename(): want %s got %s", want, got)
+	}
+}
+
+func TestMessageFilenames(t *testing.T) {
+	db, err := Open(dbPath, DBReadOnly)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	qs := "id:20091117232137.GA7669@griffis1.net"
+	threads, err := db.NewQuery(qs).Threads()
+	if err != nil {
+		t.Fatalf("error getting the threads: %s", err)
+	}
+	thread := &Thread{}
+	if !threads.Next(thread) {
+		t.Fatalf("threads.Next(thread): unable to fetch the first and only thread")
+	}
+	msgs := thread.Messages()
+	msg := &Message{}
+	if !msgs.Next(msg) {
+		t.Fatalf("msgs.Next(msg): unable to fetch the first message in the thread")
+	}
+
+	var count int
+	var fn string
+	fns := msg.Filenames()
+	for fns.Next(&fn) {
+		count++
+	}
+
+	if want, got := 2, count; want != got {
+		t.Errorf("msg.Filenames(): want %d filename got %d", want, got)
+	}
+}
+
+func TestMessageDate(t *testing.T) {
+	db, err := Open(dbPath, DBReadOnly)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	qs := "id:20091117232137.GA7669@griffis1.net"
+	threads, err := db.NewQuery(qs).Threads()
+	if err != nil {
+		t.Fatalf("error getting the threads: %s", err)
+	}
+	thread := &Thread{}
+	if !threads.Next(thread) {
+		t.Fatalf("threads.Next(thread): unable to fetch the first and only thread")
+	}
+	msgs := thread.Messages()
+	msg := &Message{}
+	if !msgs.Next(msg) {
+		t.Fatalf("msgs.Next(msg): unable to fetch the first message in the thread")
+	}
+	if want, got := time.Unix(1258500098, 0), msg.Date(); want.Unix() != got.Unix() {
+		t.Errorf("msg.Date(): want %s got %s", want, got)
+	}
+}
+
+func TestMessageHeader(t *testing.T) {
+	db, err := Open(dbPath, DBReadOnly)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	qs := "id:1258471718-6781-2-git-send-email-dottedmag@dottedmag.net"
+	threads, err := db.NewQuery(qs).Threads()
+	if err != nil {
+		t.Fatalf("error getting the threads: %s", err)
+	}
+	thread := &Thread{}
+	if !threads.Next(thread) {
+		t.Fatalf("threads.Next(thread): unable to fetch the first and only thread")
+	}
+	msgs := thread.Messages()
+	msg := &Message{}
+	for msgs.Next(msg) {
+		if msg.ID() == "1258471718-6781-2-git-send-email-dottedmag@dottedmag.net" {
+			break
+		}
+	}
+	for _, hn := range []string{"References", "references"} {
+		if want, got := "<1258471718-6781-1-git-send-email-dottedmag@dottedmag.net>", msg.Header(hn); want != got {
+			t.Errorf("msg.Header(%q): want %s got %s", hn, want, got)
+		}
+	}
+}
+
+func TestMessageTags(t *testing.T) {
+	db, err := Open(dbPath, DBReadOnly)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	qs := "id:1258471718-6781-2-git-send-email-dottedmag@dottedmag.net"
+	threads, err := db.NewQuery(qs).Threads()
+	if err != nil {
+		t.Fatalf("error getting the threads: %s", err)
+	}
+	thread := &Thread{}
+	if !threads.Next(thread) {
+		t.Fatalf("threads.Next(thread): unable to fetch the first and only thread")
+	}
```

Comment:
This same ~10 lines of code is repeated over and over again. I'm thinking it might be prudent to wrap it in a function of its own.